apt-hook/hook.cc

```diff
@@ -152,30 +153,34 @@ static void check_esm_upgrade(pkgCache::PkgIterator pkg, pkgPolicy *policy, resu
       {
          if (pf.File().Archive() != 0 && pf.File().Origin() == std::string("UbuntuESM"))
          {
-            res.esm_i_packages.push_back(pkg.Name());
-
-            // Pin-Priority: never unauthenticated APT repos == -32768
-            if (policy->GetPriority(pf.File()) == -32768)
-            {
-               res.disabled_esms_i++;
-            }
-            else
-            {
-               res.enabled_esms_i++;
+            if (std::find(res.esm_i_packages.begin(), res.esm_i_packages.end(), pkg.Name()) == res.esm_i_packages.end()) {
```

Comment:
This looks correct. It does technically make our counting algorithm O(n^2)
We might be able to fix this problem by just returning early from this function when we find that the package meets one of our criteria, but that feels like a riskier change whereas this is clearly doing what we want.
And our `n` is so small anyway..
I think this is good

Reply:
Very good point. I've added a followup issue for improvement. When we aren't under time pressure. https://github.com/canonical/ubuntu-advantage-client/issues/1581